arbitrator/src/fixed_priority_cost_function.cpp

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2019 LEIDOS.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+#include "fixed_priority_cost_function.hpp"
+#include "arbitrator_utils.hpp"
+#include "cav_msgs/ManeuverParameters.h"
+#include <limits>
+
+namespace arbitrator
+{
+    FixedPriorityCostFunction::FixedPriorityCostFunction(std::map<std::string, double> plugin_priorities) 
+    {
+        // Identify the highest priority values present in the list
+        double max_priority = std::numeric_limits<double>::lowest();
+        for (auto it = plugin_priorities.begin(); it != plugin_priorities.end(); it++)
+        {
+            if (it->second > max_priority)
+            {
+                max_priority = it->second;
+            }
+        }
+
+        // Normalize the list and invert into costs
+        for (auto it = plugin_priorities.begin(); it != plugin_priorities.end(); it++)
+        {
+            plugin_costs_[it->first] = 1.0 - (it->second / max_priority);
```

Comment:
Will be a potential "Divide by Zero" error here if all priorities are not set and default to zero? This also apply to the following ```compute_cost_per_unit_distance``` function.

Reply:
Yeah, it would be. The alternative to throwing a divide by zero exception here would be to catch the exception (or check if it would be thrown) and then throw another exception though, so I'm not clear if that's any better than just throwing a divide by zero.

Reply:
I think the easy way to handle it is to check if max_priority == 0 and assign its cost to infinity/maximum.